Runtime/Model/Breadcrumbs/Storage/BacktraceStorageLogManager.cs

```diff
@@ -0,0 +1,309 @@
+            if (attributes != null && attributes.Count > 0)
+            {
+                jsonObject.Add("attributes", new BacktraceJObject(attributes));
+            }
+            return jsonObject;
+        }
+        /// <summary>
+        /// Append breadcrumb JSON to the end of the breadcrumbs file.
+        /// </summary>
+        /// <param name="bytes">Bytes that represents single JSON object with breadcrumb informatiom</param>
+        private bool AppendBreadcrumb(byte[] bytes)
+        {
+            // size of the breadcrumb - it's negative at the beginning because we're removing 2 bytes on start
+            long appendingSize = EndOfDocument.Length * -1;
+            using (var breadcrumbStream = BreadcrumbFile.GetWriteStream())
+            {
+                //back to position before end of the document \n}
+                breadcrumbStream.Position = breadcrumbStream.Length - EndOfDocument.Length;
+
+                // append ,\n when we're appending new row to existing list of rows. If this is first row
+                // ignore it
+                if (_emptyFile == false)
+                {
+                    breadcrumbStream.Write(NewRow, 0, NewRow.Length);
+                    appendingSize += NewRow.Length;
+                }
+                else
+                {
+                    _emptyFile = false;
+                }
+                // append breadcrumbs json
+                breadcrumbStream.Write(bytes, 0, bytes.Length);
+                // and close JSON document
+                breadcrumbStream.Write(EndOfDocument, 0, EndOfDocument.Length);
+                appendingSize += (bytes.Length + EndOfDocument.Length);
+            }
+            currentSize += appendingSize;
+            _logSize.Enqueue(bytes.Length);
+            return true;
+        }
+
+        /// <summary>
+        /// Remove last n logs from the breadcrumbs file. When breacrumbs file hit 
+        /// the file size limit, this method will clear up the oldest logs to decrease
+        /// file size.
+        /// </summary>
+        private void ClearOldLogs()
+        {
+            var startPosition = GetNextStartPosition();
+            using (var breadcrumbsStream = BreadcrumbFile.GetIOStream())
+            {
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    var size = breadcrumbsStream.Length - startPosition;
+                    breadcrumbsStream.Seek(size * -1, SeekOrigin.End);
+
+                    breadcrumbsStream.CopyTo(ms);
+                    breadcrumbsStream.SetLength(size + StartOfDocument.Length);
+
+                    ms.Position = 0;
+                    breadcrumbsStream.Position = 0;
+
+                    breadcrumbsStream.Write(StartOfDocument, 0, StartOfDocument.Length);
+                    ms.CopyTo(breadcrumbsStream);
+                }
+            }
+            // decrease a size of the breadcrumb file after removing n breadcrumbs
```

Comment:
Typo: decrease a size -> decrease the size

Reply:
done thx.